mdx-models/src/main/java/com/mx/path/model/mdx/model/MdxBase.java

```diff
@@ -37,4 +36,18 @@ public final void appendWarning(Warning warning) {
     }
     this.warnings.add(warning);
   }
+
+  public final void setUserId(String userId) {
+    // Don't set the user ID if the model is annotated with MdxNested. Cache result to avoid reflection overhead.
+    if (!INCLUDE_USER_ID_MAP.containsKey(this.getClass())) {
+      MdxNested annotation = this.getClass().getAnnotation(MdxNested.class);
```

Comment:
Is there a reason `getAnnotation()` is preferred to `isAnnotationPresent()`? It looks like `isAnnotationPresent()` just calls `getAnnotation()` under the hood and does a null check. If the performance is the same, I think `isAnnotationPresent()` is more readable.

Reply:
I agree. This was just a holdover from the previous implementation. Fixing.

Reply:
This is fixed.